### PR TITLE
[Refactor] Move the library status/filters out of the GlobalContext into a LibraryContext

### PR DESCRIPTION
**Pull request by arielj**
I started working on implementing some redesign for the library and noticed the current state management of the library is spread all over the place:
- most state is in GlobalState
- some state is in Library
- some state is in the children of the library

The state of the library doesn't need to be global, moving that out saves extra re-renderings, like re-rendering the sidebar when the search changes. Also, decoupling the different components will make it easier for me to implement some of the design changes for the filtering. It also makes the code more consistent, ActionIcons was updating the local storage of the browser to store the values of some filters but the other filters were updated by GlobalState.

Now all these:
- search text
- platform filter
- store filter
- sorting
- layout
- visibility filter

are handled by the Library component and shared with a context to also avoid prop drilling

One more thing I noticed and fixed: the platform filter and the toggle to show/hide non-available games were not getting stored

this also makes the global state simpler for whenever we move out of the global context into a state store

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/frontend/components/UI/ActionIcons/index.tsx [hunk 1]
```diff
@@ -14,55 +14,38 @@ import {
...
-
-  const toggleShowFavourites = useCallback(() => {
-    setShowFavourites(!showFavourites)
-  }, [showFavourites])
+    setShowNonAvailable,
+    sortDescending,
+    setSortDescending,
+    sortInstalled,
+    setSortInstalled
+  } = useContext(LibraryContext)
```
Comment (arielj) [inline src/frontend/components/UI/ActionIcons/index.tsx:48]: some state came from the global context, some from props, now it's all from the library context

File: src/frontend/state/GlobalState.tsx [hunk 2]
```diff
@@ -673,11 +637,8 @@ class GlobalState extends PureComponent<Props> {
 
   async componentDidMount() {
     const { t } = this.props
-    const { epic, gameUpdates = [], libraryStatus, category } = this.state
-    const oldCategory: string = category
-    if (oldCategory === 'epic') {
-      this.handleCategory('all')
-    }
```
Comment (arielj) [inline src/frontend/state/GlobalState.tsx:680]: more than a year ago we migrated the `epic` filter to be called `legendary`, by now any user should already have the correct store filter saved in the local storage so we can remove this migration workaround

**Comment by flavioislima** [comment]
Could not find any other issue.

**Review by flavioislima — CHANGES_REQUESTED** [review]
Only issue is with the favorites.

**Comment by arielj** [comment]
> Testing this one here now. One issue I found is that the favorites are not being updated until you restart the app. First I added a few games to it and then clicking on the favorites button was showing nothing. Then After going to the General settings and changing the first lane to 'Favorites' made it work. But then after adding more favorites won't work until restart. image

thanks for testing, just pushed a fix for that

**Review by flavioislima — APPROVED** [review]
Works perfectly now. I noticed a bit of performance improvement as well when changing filters, layout, etc.
Maybe we won't need mobx or recoil in the short term anymore? 👀 
something to think about.

**Pull request merged**